alibi_detect/saving/validators.py

```diff
@@ -0,0 +1,84 @@
+import sys
+from typing import Any, Generic, Optional, Type, TypeVar, Union, List
+import numpy as np
+from numpy.lib import NumpyVersion
+from pydantic.fields import ModelField
+
+from alibi_detect.utils.frameworks import has_tensorflow, has_pytorch, has_keops, Framework
+
+if has_tensorflow:
+    import tensorflow as tf
+if has_pytorch:
+    import torch
+
+
+def coerce_int2list(value: int) -> List[int]:
+    """Validator to coerce int to list (pydantic doesn't do this by default)."""
+    if not isinstance(value, list):
+        return [value]
+    else:
+        return value
+
+
+# Framework validator (validates `flavour` and `backend` fields)
+def validate_framework(framework: str, field: ModelField) -> str:
```

Comment:
Is `ModelField` something to do with code completion?

Reply:
Yep. In the code-snippet in [this section](https://pydantic-docs.helpmanual.io/usage/types/#generic-classes-as-types) of the pydantics docs they suggest including `ModelField` for code completion.